pkg/client/client.go

```diff
@@ -261,3 +271,22 @@ func MergeAnnotations(toolchainObject client.Object, newAnnotations map[string]s
 	}
 	toolchainObject.SetAnnotations(annotations)
 }
+
+// ApplyUnstructuredObjects applies the given Unstructured objects on the cluster.
+func ApplyUnstructuredObjects(ctx context.Context, cl client.Client, unstructuredObjects []*unstructured.Unstructured, newLabels map[string]string) (bool, error) {
+	applyClient := NewApplyClient(cl)
+	anyApplied := false
+
+	for _, unstructuredObj := range unstructuredObjects {
+		var object = unstructuredObj.DeepCopyObject().(client.Object)
+		log.Info("applying object", "object_namespace", object.GetNamespace(), "object_name", object.GetObjectKind().GroupVersionKind().Kind+"/"+object.GetName())
+		MergeLabels(object, newLabels)
+		_, err := applyClient.ApplyObject(ctx, object)
+		if err != nil {
+			return anyApplied, err
+		}
+		anyApplied = true
```

Comment:
btw, in this case the `anyApplied` field doesn't make much sense - it will be always true as soon as there is at least one object in the given slice. There is no other option

Reply:
if it fails while applying the first object then it returns `false, err`. I mean if it enters the error case at line 285 on the first iteration.

Reply:
Maybe I should get the bool from `applyClient.ApplyObject` at line 283 ? 🤔

Reply:
if it fails with error, then we return an error anyway. 
Let's either drop it or use the boolean returned from `ApplyObject`

Reply:
yeah let's drop it for now, and we can introduce it if needed later.

See https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/34595ac1a48392faea9925e454e62c37501512e8 

Thanks for the suggestion.